experimenter/tests/integration/nimbus/pages/demo_app/frontend.py

```diff
@@ -0,0 +1,47 @@
+import time
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from nimbus.pages.base import Base
+
+
+class DemoAppPage(Base):
+    def __init__(self, driver):
+        self.driver = driver
+
+    def navigate_to(self):
+        demo_app_url = "http://demo-app-frontend:3000/"
+        self.driver.get(demo_app_url)
+
+    def wait_for_result_text(self, text_list):
+        xpath_conditions = " | ".join([f".='{text}'" for text in text_list])
+        xpath = f"//h1[{' or '.join(xpath_conditions.split(' | '))}]"
+
+        return WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.XPATH, xpath))
+        )
+
+    def fill_and_send_form_data(self, client_id, context):
+        time.sleep(10)
+        client_id_input = WebDriverWait(self.driver, 10).until(
```

Comment:
okay, the reason why I have added sleep specifically is that I do want to wait some seconds and then fill in the client id and context, The reason is that things are happening so fast, when we launch the experiment, the experiment pushes to RS, gets approve and immediately frontend start hitting the cirrus, in some cases, Cirrus is not able to get the copy of the latest recipes and respond accordingly, On cirrus side sync time is 1 seconds (minimum value that I could set, meaning cirrus will contact to RS to get a new copy every second), so in some test cases we see the tests are passing and in some not. Whereas if I use wait it only guarantees that it will wait a maximum of lets say 10 seconds or 30 seconds for that element to appear, so the element is found every time, but it still needs to wait a few seconds before starting to discover the elements so that cirrus and RS can get some time to sync. Also, I have tested locally. I hope this explains, let me know your thoughts

Reply:
That works, I'm not opposed to the idea, especially if it helps.